### ISSUE TITLE
Simulation does not start. No simulated data are sending to IoT Hub.

### ISSUE THREAD

**Issue by mtheolog** [Azure/device-simulation-dotnet#218]
# Type of issue 

- [X] Bug
- [ ] New feature
- [ ] Enhancement

# Description 
I am trying to define a new simulated device by following the guidelines at "Create a new simulated device" doc of https://docs.microsoft.com/en-us/azure/iot-accelerators/iot-accelerators-remote-monitoring-test. 

At the end, none device is connected to IoT hub and thus no simulated data are sending too.
At http://localhost:9003/v1/status the following is displayed:

{"Name":"DeviceSimulation","Status":"OK:Alive and well","CurrentTime":"2018-06-13T08:35:38+00:00","StartTime":"2018-06-13T08:34:57+00:00","UpTime":41,"UID":"WebService.09c728e9-8cec-4f13-b81e-9f703a52e521","Properties":{"SimulationRunning":"false","PreprovisionedIoTHub":"true","ActiveDevicesCount":"0","TotalMessagesCount":"0","FailedMessagesCount":"0","MessagesPerSecond":"0.00","FailedDeviceConnectionsCount":"0","FailedDeviceTwinUpdatesCount":"0","SimulationErrorsCount":"0"},"Dependencies":{"Storage":"OK:Alive and well","PreprovisionedIoTHub":"OK"},"$metadata":{"$type":"Status;1","$uri":"/v1/status"}}

It seems that the simulation is not started at all. But in the wiki is specified that "Simulations start immediately, unless specified differently".

# Steps to reproduce 

1. Follow  https://docs.microsoft.com/en-us/azure/iot-accelerators/iot-accelerators-remote-monitoring-test. 

# Expected behavior 

Simulated data sent to IoT Hub

# Current behavior 
...

# Possible solution 
...

# Context and Environment
* Operating System: 
* GitHub branch: 
* .NET Runtime: 

# Code to reproduce the bug 
```
using Xunit;

public class MyTest
{
    [Fact]
    public void TestToReproduceIssue()
    {
        // Arrange
        ...

        // Act
        ...

        // Assert
        ...
    }
}
```

**Comment by mtheolog** [comment]
I've managed to start the simulation (in the simulation.1 at Data field i changed the value of the 'enabled' from false to true).

BUT still none device is connected to IoT Hub and thus no telemetry are sending. I enclose the following portions of the log file

[DEBUG][2018-06-14 12:25:39.118][WebService.f0986ffb-0838-41db-89de-1cd409afd3ae][DeviceTelemetryActor:Run] ReadyToStart, {"deviceId":"truck-01.0"}
[DEBUG][2018-06-14 12:25:39.119][WebService.f0986ffb-0838-41db-89de-1cd409afd3ae][DeviceTelemetryActor:Run] ReadyToStart, {"deviceId":"truck-02.0"}
[DEBUG][2018-06-14 12:25:39.120][WebService.f0986ffb-0838-41db-89de-1cd409afd3ae][DeviceTelemetryActor:Run] ReadyToStart, {"deviceId":"chiller-02.0"}
**[INFO][2018-06-14 12:25:39.120][WebService.f0986ffb-0838-41db-89de-1cd409afd3ae][SimulationRunner:SendTelemetryThread] Telemetry loop completed, {"durationMsecs":7,"stats":{"nothingToDo":0,"device not connected yet":10}}**
[DEBUG][2018-06-14 12:25:39.308][WebService.f0986ffb-0838-41db-89de-1cd409afd3ae][DeviceConnectionActor:Run] ReadyToFetch, {"deviceId":"elevator-02.0"}
[DEBUG][2018-06-14 12:25:39.308][WebService.f0986ffb-0838-41db-89de-1cd409afd3ae][Fetch:RunAsync] Fetching device..., {"deviceId":"elevator-02.0"}

[DEBUG][2018-06-14 12:25:39.701][WebService.f0986ffb-0838-41db-89de-1cd409afd3ae][DeviceTelemetryActor:Run] ReadyToStart, {"deviceId":"engine-01.0"}
[DEBUG][2018-06-14 12:25:39.701][WebService.f0986ffb-0838-41db-89de-1cd409afd3ae][DeviceTelemetryActor:Run] ReadyToStart, {"deviceId":"truck-01.0"}
[DEBUG][2018-06-14 12:25:39.720][WebService.f0986ffb-0838-41db-89de-1cd409afd3ae][DeviceTelemetryActor:Run] ReadyToStart, {"deviceId":"truck-02.0"}
[DEBUG][2018-06-14 12:25:39.720][WebService.f0986ffb-0838-41db-89de-1cd409afd3ae][DeviceTelemetryActor:Run] ReadyToStart, {"deviceId":"chiller-02.0"}
[INFO][2018-06-14 12:25:39.737][WebService.f0986ffb-0838-41db-89de-1cd409afd3ae][SimulationRunner:SendTelemetryThread] Telemetry loop completed, {"durationMsecs":101,"stats":{"nothingToDo":0,"device not connected yet":10}}
[ERROR][2018-06-14 12:25:39.758][WebService.f0986ffb-0838-41db-89de-1cd409afd3ae][Fetch:RunAsync] External dependency error while fetching the device, {"deviceId":"truck-01.0","e":{"ExceptionFullName":"Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Exceptions.ExternalDependencyException","ExceptionMessage":"Unable to fetch the IoT device","StackTrace":"   at Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Devices.<GetAsync>d__15.MoveNext() in C:\\Users\\mtheo\\source\\repos\\IoT-Accelarator\\device-simulation-dotnet-master\\Services\\Devices.cs:line 171\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()\r\n   at Microsoft.Azure.IoTSolutions.DeviceSimulation.SimulationAgent.DeviceConnection.Fetch.<RunAsync>d__6.MoveNext() in C:\\Users\\mtheo\\source\\repos\\IoT-Accelarator\\device-simulation-dotnet-master\\SimulationAgent\\DeviceConnection\\Fetch.cs:line 41","Source":"Microsoft.Azure.IoTSolutions.DeviceSimulation.Services","Data":{},"InnerException":null}}
[INFO][2018-06-14 12:25:39.767][WebService.f0986ffb-0838-41db-89de-1cd409afd3ae][RatedCounter:LogThroughput] Registry operations, {"speed":5.989583333333333}
[ERROR][2018-06-14 12:25:39.760][WebService.f0986ffb-0838-41db-89de-1cd409afd3ae][Fetch:RunAsync] External dependency error while fetching the device, {"deviceId":"chiller-02.0","e":{"ExceptionFullName":"Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Exceptions.ExternalDependencyException","ExceptionMessage":"Unable to fetch the IoT device","StackTrace":"   at Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Devices.<GetAsync>d__15.MoveNext() in C:\\Users\\mtheo\\source\\repos\\IoT-Accelarator\\device-simulation-dotnet-master\\Services\\Devices.cs:line 171\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()\r\n   at Microsoft.Azure.IoTSolutions.DeviceSimulation.SimulationAgent.DeviceConnection.Fetch.<RunAsync>d__6.MoveNext() in C:\\Users\\mtheo\\source\\repos\\IoT-Accelarator\\device-simulation-dotnet-master\\SimulationAgent\\DeviceConnection\\Fetch.cs:line 41","Source":"Microsoft.Azure.IoTSolutions.DeviceSimulation.Services","Data":{},"InnerException":null}}
**[ERROR][2018-06-14 12:25:39.760][WebService.f0986ffb-0838-41db-89de-1cd409afd3ae][Fetch:RunAsync] External dependency error while fetching the device, {"deviceId":"elevator-02.0","e":{"ExceptionFullName":"Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Exceptions.ExternalDependencyException","ExceptionMessage":"Unable to fetch the IoT device","StackTrace":"   at Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Devices.<GetAsync>d__15.MoveNext() in C:\\Users\\mtheo\\source\\repos\\IoT-Accelarator\\device-simulation-dotnet-master\\Services\\Devices.cs:line 171\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()\r\n   at Microsoft.Azure.IoTSolutions.DeviceSimulation.SimulationAgent.DeviceConnection.Fetch.<RunAsync>d__6.MoveNext() in C:\\Users\\mtheo\\source\\repos\\IoT-Accelarator\\device-simulation-dotnet-master\\SimulationAgent\\DeviceConnection\\Fetch.cs:line 41","Source":"Microsoft.Azure.IoTSolutions.DeviceSimulation.Services","Data":{},"InnerException":null}}**
[DEBUG][2018-06-14 12:25:39.767][WebService.f0986ffb-0838-41db-89de-1cd409afd3ae][DeviceConnectionActor:ScheduleFetch] Device fetch scheduled, {"deviceId":"truck-01.0","Status":"ReadyToFetch","When":"2018-06-14 12:25:39.767"}
[INFO][2018-06-14 12:25:39.770][WebService.f0986ffb-0838-41db-89de-1cd409afd3ae][RatedCounter:LogThroughput] Registry operations, {"speed":4.9947970863683659}

Please let me know how can i proceed.

Regards, Maria.

PS. As I have to connect our physical devices to azure remote monitoring and thus to define our device models, I wonder whether i can by-pass the device simulation?  Is there any other way to define custom device models?

**Comment by ppathan** [comment]
@hathind-ms Can you want to take a look at this?

**Comment by hathind-ms** [comment]
Hi @mtheolog,


Thank you for reporting the issue. I would need some more details in order to assist you with your issue. Can you please answer the questions below?

1.	How did you deploy the solution?
2.	Did the simulation ever start? If so, did you stop it?
3.	How many devices did you create for your simulation?
4.	Are you running any other application on your IoT Hub?
5.	Have you tried re-deploying the solution?

Also, can you please share complete log?

**Comment by TameezM** [comment]
I got the same issue. i'm working on it to figure out whats the issue, i followed the steps given in [https://docs.microsoft.com/en-us/azure/iot-accelerators/iot-accelerators-remote-monitoring-test](https://docs.microsoft.com/en-us/azure/iot-accelerators/iot-accelerators-remote-monitoring-test)
i'm missing something obvious which is not visible to my eyes, 
following is my output in browser at dotnet simulator service
`{"Name":"DeviceSimulation","Status":"OK:Alive and well","CurrentTime":"2018-08-02T09:09:00+00:00","StartTime":"2018-08-02T08:56:25+00:00","UpTime":755,"UID":"WebService.82d59c22-7e7b-4fae-b637-54bd3b012904","Properties":{"SimulationRunning":"true","PreprovisionedIoTHub":"true","PreprovisionedIoTHubInUse":"true","PreprovisionedIoTHubMetricsUrl":"https://portal.azure.com/undefined.onmicrosoft.com#resource/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/undefined/providers/Microsoft.Devices/IotHubs/undefined/Metrics","ActiveDevicesCount":"2","TotalMessagesCount":"0","FailedMessagesCount":"0","MessagesPerSecond":"0.00","FailedDeviceConnectionsCount":"0","FailedDeviceTwinUpdatesCount":"0","SimulationErrorsCount":"1279"},"Dependencies":{"Storage":"OK:Alive and well","PreprovisionedIoTHub":"OK"},"$metadata":{"$type":"Status;1","$uri":"/v1/status"}}`

**Comment by hathind-ms** [comment]
@TameezM - Can you please share some logs from your VM? Please find instructions to obtain logs here: [https://github.com/Azure/device-simulation-dotnet/wiki/Simulation-Service-Developer-Reference-Guide#viewing-logs](https://github.com/Azure/device-simulation-dotnet/wiki/Simulation-Service-Developer-Reference-Guide#viewing-logs )

**Comment by peterfelts** [comment]
@TameezM, this part is interesting (below) where we've got a null subscription ID, a null tenant name, and a null resource group. Have you tried a new deployment? We just pushed changes to master on 11/8/2018.

{"SimulationRunning":"true","PreprovisionedIoTHub":"true","PreprovisionedIoTHubInUse":"true","PreprovisionedIoTHubMetricsUrl":"https://portal.azure.com/undefined.onmicrosoft.com#resource/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/undefined/providers/Microsoft.Devices/IotHubs/undefined/Metrics","ActiveDevicesCount":"2","TotalMessagesCount":"0","FailedMessagesCount":"0","MessagesPerSecond":"0.00","FailedDeviceConnectionsCount":"0","FailedDeviceTwinUpdatesCount":"0","SimulationErrorsCount":"1279"},"Dependencies":{"Storage":"OK:Alive and well","PreprovisionedIoTHub":"OK"}
